app/models/user.rb

```diff
@@ -4,4 +4,30 @@ class User < ApplicationRecord
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def has_membership_application?
+    membership_applications.size > 0 && (membership_applications.select { |app| app.persisted? }).count > 0
```

Comment:
The `membership_applications` collection will (if not empty) *always* have persisted members, since the tables are joined by foreign keys and the FK will be null is the record on the other side is not saved.

So, checking persistence is not useful.

So, this member could just be
```
membership_applications.size > 0
```